apps/server/src/challenges/challenges.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { Challenge } from '@/challenges/entities/challenges.entity';
+import { CreateChallengeDto } from '@/challenges/dto/challenges.dto';
+
+@Injectable()
+export class ChallengesService {
+  constructor(
+    @InjectRepository(Challenge)
+    private challengeRepository: Repository<Challenge>,
+  ) {}
+
+  /**
+   * Create a new challenge.
+   * @param {CreateChallengeDto} payload - The payload containing challenge information.
+   * @returns {Promise<ChallengeEntity>} - The created challenge entity.
+   */
+  public async create(payload: CreateChallengeDto) {
```

Comment:
We don't need to mark these as `public` as well.

Reply:
Done